<?xml version="1.0"?>
<launch>
  <param name="/robot_description" command="$(find xacro)/xacro '$(find argo_tracker_description)/urdf/argo_tracker_default.urdf.xacro'" />
  <param name="/robot_description_semantic" textfile="$(find argo_tracker_robot_moveit_config)/config/taurob_tracker.srdf" />
  
  <rosparam command="load" file="$(find argo_tracker_robot_moveit_config)/config/kinematics.yaml" ns="/robot_description_kinematics" />
  <rosparam command="load" file="$(find argo_tracker_robot_moveit_config)/config/kinematics.yaml" ns="/argo_tracker_inverse_kinematics_benchmark" />

  <node name="argo_tracker_inverse_kinematics_benchmark" pkg="moveit_kinematics_plugin_benchmark" type="inverse_kinematics_benchmark_node" output="screen">
    <param name="random_samples" value="true"/>
    <param name="num_random_samples" value="1000"/>
    <rosparam subst_value="true">
      groups:
         #- arm_group
         #- arm_with_head_group
        - arm_with_gripper_group_trac
        - arm_with_gripper_group_ceres
      sample_states:
        - [0.627938, 1.23032, 4.59099, 2.55584, 2.40853, -1.90034]        
    </rosparam>
  </node>
</launch>
